src/modules/RA/Scene/components/GeolayerSelect.js

```diff
@@ -16,6 +20,49 @@ const sanitizeRestProps = ({
   ...rest
 }) => rest;
 
+const PER_PAGE = 100;
+const MAX_NON_NATIVE_SELECT = 500;
+
+/**
+ * Get all layers accross pagination.
+ * @param {*} dataProvider The dataproviver instance
+ * @param {*} filter optionnal filter
+ */
+const getAllPaginatedLayer = async (dataProvider, mounted, filter = {}) => {
+  // Get first layer to know total
+  const { total, data: dataFirstPage } = await dataProvider(GET_LIST, RES_DATALAYER, {
+    pagination: { page: 1, perPage: PER_PAGE },
+    sort: { field: 'name', order: 'ASC' },
+    filter,
+  });
+
+  const pageCount = Math.ceil(total / PER_PAGE);
+
+  const result = Array.from({ length: pageCount });
+  // Keep first result
+  result[0] = dataFirstPage;
+
+  // Query all next layers
+  const promises = Array.from({ length: pageCount - 1 }).map(
+    (_, index) => dataProvider(GET_LIST, RES_DATALAYER, {
+      pagination: { page: index + 2, perPage: PER_PAGE },
+      sort: { field: 'name', order: 'ASC' },
+      filter,
+    }).then(({ data }) => {
+      result[index + 1] = data;
+    }),
+  );
+
+  await Promise.all(promises);
```

Comment:
Another pattern suggestion for this whole process:

```jsx
const results = [
  ...dataFirstPage,
  ...(await Promise.all(promises),)
];
```

Reply:
Done